server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/chatDb/chatDb.ts

```diff
@@ -611,20 +612,57 @@ export class ChatDatabase {
         }
     }
 
-    private handleToolUses(messages: Message[], newUserMessage: ChatMessage): void {
+    private validateToolUses(messages: Message[], newUserMessage: ChatMessage): boolean {
         if (messages.length === 0) {
             if (newUserMessage.userInputMessage?.userInputMessageContext?.toolResults) {
                 this.#features.logging.debug('No history message found, but new user message has tool results.')
                 newUserMessage.userInputMessage.userInputMessageContext.toolResults = undefined
                 // tool results are empty, so content must not be empty
                 newUserMessage.userInputMessage.content = 'Conversation history was too large, so it was cleared.'
             }
-            return
+            return true
         }
 
         const lastMsg = messages[messages.length - 1]
+        const toolResults = newUserMessage.userInputMessage?.userInputMessageContext?.toolResults
+
+        if (toolResults && toolResults.length > 0) {
+            // If last message has no tool uses but new message has tool results, this is invalid
+            if (!lastMsg.toolUses || lastMsg.toolUses.length === 0) {
+                this.#features.logging.warn('New message has tool results but last message has no tool uses')
```

Comment:
will warn show up?

Reply:
From my testing it shows up: `[Warn  - 7:14:04 PM] [2025-05-07T02:14:04.645Z] testing log level`.

Since we show info and logging levels are hierarchical I think we should see them.